coderd/templates.go

```diff
@@ -400,29 +406,36 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	httpapi.Write(rw, http.StatusOK, convertTemplate(updated, count))
+	httpapi.Write(rw, http.StatusOK, convertTemplate(r.Context(), api.Database, updated, count))
 }
 
-func convertTemplates(templates []database.Template, workspaceCounts []database.GetWorkspaceOwnerCountsByTemplateIDsRow) []codersdk.Template {
+func convertTemplates(ctx context.Context, db database.Store, templates []database.Template, workspaceCounts []database.GetWorkspaceOwnerCountsByTemplateIDsRow) []codersdk.Template {
 	apiTemplates := make([]codersdk.Template, 0, len(templates))
 	for _, template := range templates {
 		found := false
 		for _, workspaceCount := range workspaceCounts {
 			if workspaceCount.TemplateID.String() != template.ID.String() {
 				continue
 			}
-			apiTemplates = append(apiTemplates, convertTemplate(template, uint32(workspaceCount.Count)))
+			apiTemplates = append(apiTemplates, convertTemplate(ctx, db, template, uint32(workspaceCount.Count)))
 			found = true
 			break
 		}
 		if !found {
-			apiTemplates = append(apiTemplates, convertTemplate(template, uint32(0)))
+			apiTemplates = append(apiTemplates, convertTemplate(ctx, db, template, uint32(0)))
 		}
 	}
 	return apiTemplates
 }
 
-func convertTemplate(template database.Template, workspaceOwnerCount uint32) codersdk.Template {
+func convertTemplate(ctx context.Context, db database.Store, template database.Template, workspaceOwnerCount uint32) codersdk.Template {
+	var ownerName string
+	if template.OwnerID.Valid {
+		owner, err := db.GetUserByID(ctx, template.OwnerID.UUID)
+		if err == nil {
```

Comment:
We should ideally propagate context cancellation here somehow.

This is just an example scenario:

We receive a request, we decide it should at most take 1 second:

```go
ctx, _ := context.WithTimeout(r.Context(), 1 * time.Second)
```

(The error could also be from the DB, connection issue, statement timeout, etc.)

This context would then be passed on to `convertTemplate`(s) and as a result, all of the `ownerName`s would likely become the empty string. In this instance it's not the end of the world, but it's a hidden silent failure which isn't great. IMO we should strive to always fail correctly.

Reply:
Handled errors in the caller methods as discussed.